app/elements/batch-image-search/batch-image-search.html

```diff
@@ -0,0 +1,379 @@
+
+          <button-action id="clearAll" class="right-space" text="Clear" click-listener="[[_createClearListener()]]"></button-action>
+          <loading-spinner show="[[loading]]" type="Similar Images"></loading-spinner>
+        </div>
+        
+        <iron-ajax
+        id="xhr"
+        handle-as="json"
+        with-credentials="true"
+        loading="{{loading}}"
+        on-response="handleResponse"></iron-ajax>
+
+      </div>
+    </paper-dialog>
+
+    <paper-dialog id="resultImages" modal entry-animation="fade-in-animation" exit-animation="fade-out-animation">
+        <div class="layout vertical">
+          <div class="layout horizontal">
+            <span class="bold tall flex">Similar Images Result</span>
+            <paper-icon-button icon="clear" dialog-confirm title="Close"></paper-icon-button>
+          </div>
+          <div class="divider"></div>
+          <div class="layout vertical">
+            <span class="bold tall flex">Compared Image</span>
+              <image-thumbnail
+              source="[[compareSourceImage]]"
+              style-class="[[getBlurStyleClass(blur)]]"
+              target="_blank"
+              class="simImages"
+              text="open image">
+            </image-thumbnail>
+          </div>
+          <div class="layout">
+            <div class="divider"></div>
+            <template is="dom-repeat" items="[[images]]" rendered-item-count="{{renderedCount}}">
+              <image-thumbnail-with-link
+                link = {{displayLink(index)}}
+                source="[[item]]"
+                style-class="[[getBlurStyleClass(blur)]]"
+                target="_blank"
+                class="simImages"
+                text="open image">
+              </image-thumbnail-with-link>
+            </template>
+            <template is="dom-if" if="{{!renderedCount}}">
+              <span class="text">No Similar Images </span>
+            </template> 
+          </div>
+        </div>
+
+      </paper-dialog>
+  </template>
+
+  <script>
+  (function() {
+    'use strict';
+
+    Polymer({
+      is: 'batch-image-search',
+
+      properties: {
+        // memex Username
+        memexUser: {
+          type: String,
+          notify: true,
+          value: ''
+        },
+        // memex Password
+        memexPassword: {
+          type: String,
+          notify: true,
+          value: ''
+        },
+        // Type of image - either url or base64
+        imageType: {
+          type: String,
+          value: ''
+        },
+        //Store all Images
+        images: {
+          type: Array,
+          value: []
+        },
+        // Store all sha1 of the images result
+        sha1Images: {
+          type: Array,
+          value: []
+        },
+        // Count of rendered results
+        renderedCount: {
+          type: Number,
+          value: 0
+        },
+        // enable or disable search button
+        boolButton: {
```

Comment:
This property name could be more descriptive.  Maybe something like `enableSearch`?